x-pack/plugins/observability_solution/infra/common/alerting/metrics/alert_link.ts

```diff
@@ -143,12 +150,18 @@ export const getMetricsViewInAppUrl = ({
   }, {} as Record<string, InventoryItemType>);
 
   // detemines if the groupBy has a field that the asset details supports
-  const supportedAssetId = groupBy?.find((field) => !!assetTypeByAssetId[field]);
+  const supportedAssetId = groupBy.find((field) => !!assetTypeByAssetId[field]);
   // assigns a nodeType if the groupBy field is supported by asset details
   const supportedAssetType = supportedAssetId ? assetTypeByAssetId[supportedAssetId] : undefined;
 
   if (supportedAssetType) {
     const assetId = fields[findInventoryModel(supportedAssetType).fields.id];
+
+    // Pathological case found within tests. If no assetId, just redirect.
```

Comment:
q: do we need this comment? IMO, validating the existence of `assetId` is important regardless of the test. wdyt?

Reply:
Maybe should be better worded, but usually, if I come across cases that are not immediately obvious (despite being tested in code), I prefer leaving a comment explaining that case in order for it to be less confusing to myself later on and others. Tests cover assertions/expectations, comments should cover rationale/reasoning.

Else, I can just remove it if you feel it isn't necessary.

Reply:
Makes sense. Maybe `If no assetId, just redirect to metrics explorer` makes this line clear enough then? It's just a suggestion. I'll leave it up to you to decide what's best.